### PR TITLE
[SPARK-45330][BUILD ] Upgrade ammonite to 2.5.11

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to upgrade ammonite to 2.5.11 

### Why are the changes needed?
This is a version that supports Scala 2.13.12.
 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -228,7 +228,7 @@
     ./python/pyspark/sql/pandas/utils.py, and ./python/setup.py too.
     -->
     <arrow.version>13.0.0</arrow.version>
-    <ammonite.version>2.5.9</ammonite.version>
+    <ammonite.version>2.5.9-2-7633e640</ammonite.version>
```
Comment (LuciferYang) [inline pom.xml:231]: cc @dongjoon-hyun This is the `ammonite` version that supports Scala 2.13.12, it seems the tests have passed, we can wait for an official release of the `ammonite` version.

Comment (dongjoon-hyun) [reply pom.xml:231]: Thank you, @LuciferYang ! Please create a JIRA issue and link to this PR.

Comment (LuciferYang) [reply pom.xml:231]: created SPARK-45330 and SPARK-45331, I will update this PR after ammonite 2.6.0 is released.

**Comment by LuciferYang** [comment]
test amm 2.5.11 with Scala 2.13.12 passed

https://github.com/LuciferYang/spark/runs/17221539327

image


will revert to test Scala 2.13.11, let's upgrade amm first

**Review by srowen — APPROVED** [review]
Looks OK pending tests

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you!
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @srowen and @dongjoon-hyun

**Pull request closed without merging**